packages/compiler-core/src/codegen.ts

```diff
@@ -533,8 +533,9 @@ function genHoists(hoists: (JSChildNode | null)[], context: CodegenContext) {
 
   // generate inlined withScopeId helper
   if (genScopeId) {
+    const functionDeclaration = context.isTS ? '(n: () => any)' : 'n'
```

Comment:
```suggestion
    const functionDeclaration = context.isTS ? '(n: any)' : 'n'
```

Maybe `any` is more suitable here?
Because `n` is also used as the return value of `_withScopeId`, the inferred `() => any` doesn't fit.